### ISSUE TITLE
Support for Linux GCC 4.8.2

### ISSUE THREAD

**Issue by zhoub** [j-cube/multiverse#7]
@panta Hey Marco, I made some changes to compile the library with GCC 4.8.2 on Linux, please take a look the following diff and act accordingly, thanks! Contact me for anything.

``` c++
diff --git a/lib/Alembic/AbcCoreGit/Git.cpp b/lib/Alembic/AbcCoreGit/Git.cpp
index 0e17ebd..e0cfd57 100644
--- a/lib/Alembic/AbcCoreGit/Git.cpp
+++ b/lib/Alembic/AbcCoreGit/Git.cpp
@@ -259,7 +259,7 @@ void LibGit::CleanUp()

 std::ostream& operator<< (std::ostream& out, const GitMode& value)
 {
-    switch (value)
+    switch (GitMode::Type(value))
     {
     case GitMode::Read:
         out << "R";
diff --git a/lib/Alembic/AbcCoreGit/SampleStore.cpp b/lib/Alembic/AbcCoreGit/SampleStore.cpp
index 71f67d5..2104fd6 100644
--- a/lib/Alembic/AbcCoreGit/SampleStore.cpp
+++ b/lib/Alembic/AbcCoreGit/SampleStore.cpp
@@ -404,7 +404,7 @@ void TypedSampleStore<std::string>::checkSamplePieceForAdd( const std::string& i
 template <>
 void TypedSampleStore<std::wstring>::checkSamplePieceForAdd( const std::wstring& iSamp, const AbcA::ArraySample::Key& key, const AbcA::Dimensions& dims )
 {
-    ABCA_ASSERT( iSamp.find( '\0' ) == std::wstring::npos,
+    ABCA_ASSERT( iSamp.find( L'\0' ) == std::wstring::npos,
         "string contains NUL character" );
 }

diff --git a/lib/Alembic/AbcCoreGit/SampleStore.h b/lib/Alembic/AbcCoreGit/SampleStore.h
index e07ae30..587299c 100644
--- a/lib/Alembic/AbcCoreGit/SampleStore.h
+++ b/lib/Alembic/AbcCoreGit/SampleStore.h
@@ -197,12 +197,12 @@ protected:
     const std::vector<T>& data(const AbcA::ArraySample::Key& key) { return ks()->data(key); }

     bool hasIndex(size_t sampleIndex) const           { return (m_index_to_kid.count(sampleIndex) != 0); }
-    size_t sampleIndexToKid(size_t sampleIndex) const { return m_index_to_kid[sampleIndex]; }
-    size_t sampleIndexToKid(size_t sampleIndex)       { return m_index_to_kid[sampleIndex]; }
+    size_t sampleIndexToKid(size_t sampleIndex) const { return m_index_to_kid.find(sampleIndex)->second; }
+    size_t sampleIndexToKid(size_t sampleIndex)       { return m_index_to_kid.find(sampleIndex)->second; }

     bool hasDimensions(size_t kid) const                          { return (m_kid_dims.count(kid) != 0); }
-    const AbcA::Dimensions& getDimensions(size_t kid) const       { return m_kid_dims[kid]; }
-    AbcA::Dimensions getDimensions(size_t kid)                    { return m_kid_dims[kid]; }
+    const AbcA::Dimensions& getDimensions(size_t kid) const       { return m_kid_dims.find(kid)->second; }
+    AbcA::Dimensions getDimensions(size_t kid)                    { return m_kid_dims.find(kid)->second; }
     void setDimensions(size_t kid, const AbcA::Dimensions& dims)  { m_kid_dims[kid] = dims; }

     const AbcA::ArraySample::Key& sampleIndexToKey(size_t sampleIndex) const { size_t kid = sampleIndexToKid(sampleIndex); return KidToKey(kid); }

```

**Comment by zhoub** [comment]
Sorry, for the **sampleIndexToKid** my fix was wrong, the reason is that the `std::map::operator[]` is not constant but here it uses the `const` keyword.

**Comment by zhoub** [comment]
```
[Bo@hotakami mfh-gcc482]$ /opt/bin/gcc482 -v
Using built-in specs.
COLLECT_GCC=/opt/bin/gcc482
COLLECT_LTO_WRAPPER=/opt/libexec/gcc/x86_64-unknown-linux-gnu/4.8.2/lto-wrapper
Target: x86_64-unknown-linux-gnu
Configured with: ./configure --prefix=/opt --program-suffix=482 --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --disable-libunwind-exceptions --enable-__cxa_atexit
Thread model: posix
gcc version 4.8.2 (GCC)
```

**Comment by panta** [comment]
Applied @zhoub fixes in cf6fc34.
The [multiverse-build-machine](https://github.com/j-cube/multiverse-build-machine/tree/ubuntu-trusty64-gcc48-multiverse-1.5.8) and [alembic-builder](https://github.com/j-cube/alembic-builder/tree/linux-gcc48-multiverse-1.5.8) have both been updated to use ubuntu trusty and gcc 4.8.4 (that is the 4.8 available in ubuntu out of the box, but the changes should be the same as per 4.8.2) and need these fixes to build correctly.
See/use the [ubuntu-trusty64-gcc48-multiverse-1.5.8](https://github.com/j-cube/multiverse-build-machine/tree/ubuntu-trusty64-gcc48-multiverse-1.5.8) of the multiverse-build-machine.
